src/plugins/unified_histogram/public/chart/chart.tsx

```diff
@@ -109,39 +103,43 @@ export function Chart({
   input$: originalInput$,
   lensAdapters,
   lensEmbeddableOutput$,
-  isOnHistogramMode,
-  histogramQuery,
   isChartLoading,
   onChartHiddenChange,
   onTimeIntervalChange,
-  onSuggestionChange,
   onBreakdownFieldChange,
   onTotalHitsChange,
   onChartLoad,
   onFilter,
   onBrushEnd,
   withDefaultActions,
 }: ChartProps) {
+  const lensVisServiceCurrentSuggestionContext = useObservable(
+    lensVisService.currentSuggestionContext$
+  );
+  const lensAttributesContext = useObservable(lensVisService.lensAttributesContext$);
+  const allSuggestions = useObservable(lensVisService.allSuggestions$);
+  const currentSuggestion = lensVisServiceCurrentSuggestionContext?.suggestion;
+
   const [isSaveModalVisible, setIsSaveModalVisible] = useState(false);
   const [isFlyoutVisible, setIsFlyoutVisible] = useState(false);
   const { chartRef, toggleHideChart } = useChartActions({
     chart,
     onChartHiddenChange,
   });
 
-  const chartVisible = isChartAvailable && !!chart && !chart.hidden;
+  const chartVisible =
+    isChartAvailable &&
+    !!chart &&
+    !chart.hidden &&
+    !!lensAttributesContext &&
+    !!lensAttributesContext?.attributes;
 
   const input$ = useMemo(
     () => originalInput$ ?? new Subject<UnifiedHistogramInputMessage>(),
     [originalInput$]
   );
 
-  const { filters, query, getTimeRange, updateTimeRange, relativeTimeRange } = useRequestParams({
-    services,
-    query: originalQuery,
-    filters: originalFilters,
-    timeRange: originalTimeRange,
-  });
+  const { filters, query, getTimeRange, updateTimeRange, relativeTimeRange } = requestParams;
```

Comment:
I am still quite confused with the logic around useRequestParams and time ranges. I moved it up to `layout`. Maybe this is why some functional tests are failing now.